src/rez/pip.py

```diff
@@ -265,18 +262,24 @@ def pip_install_package(source_name, pip_version=None, python_version=None,
         tools = []
         src_dst_lut = {}
 
-        for installed_file in distribution.list_installed_files(allow_fail=True):
-            source_file = os.path.normpath(os.path.join(destpath, installed_file[0]))
+        for installed_file in distribution.list_installed_files():
+            if 'bin' in installed_file[0]:
```

Comment:
It's not obvious what's going on here, a comment would help. Also, I take it that this covers a case where `installed_file[0]` is present within a 'bin' dir; in this case, a substring search isn't robust enough, I would search for 'bin' in the list generated from `str.split(os.sep)`. I gather that placement of said bin dir is important also, I'd check for that as well.

Reply:
I seems like I also commented this line in https://github.com/nerdvegas/rez/pull/628/files#r296459995. I'm always confused when the same change appears in two different PRs.

Here is the specification of the database of installed python distributions: https://www.python.org/dev/peps/pep-0376/#record

Reply:
I have been looking into ways to improve the readability of this. Since this is installed as a wheel distribution, all the installed files are listed in a CSV file called RECORD including the bin scripts as such there is no worry of missing any files from the list.

I am aware of the specification for the database generated pointed out by JC but there seems to be some inconsistencies when combining the wheel install with the target flag to set a directory. Specifically, the bin script paths in the RECORD file
are wrong.

Essentially it points the bin scripts to be two directories up instead of being in the base location.

Have a look at an example install of pydocstyle library:

![image](https://user-images.githubusercontent.com/47409392/60071050-d0a40280-9753-11e9-85d0-d254becb77f6.png)

The bin folder is right there along with the rest of source files but a quick look into the RECORD file, revelas that the bin location should be somewhere else...

![image](https://user-images.githubusercontent.com/47409392/60071123-08ab4580-9754-11e9-9e2e-02c73b997f91.png)

This is why you see all that weird parsing and path combining taking place. I am trying to see if I am missing something but the rest of the path are correct starting right from the destpath location as base.

Using --prefix instead of --target will put the files in a site-packages directory and the scripts in a bin folder

![image](https://user-images.githubusercontent.com/47409392/60071330-ad2d8780-9754-11e9-8c2f-76da8eb6f8aa.png)

![image](https://user-images.githubusercontent.com/47409392/60071357-bd456700-9754-11e9-8ca0-413b25323d61.png)

This time the RECORD file points the bin scripts to be 3 directories up from base.

![image](https://user-images.githubusercontent.com/47409392/60071423-fe3d7b80-9754-11e9-8457-3bdff413ad0c.png)

which is strangely correct but doing so will require to parse the source packages out of lib/python/site-packages..

I will keep looking into for a bit longer and see if I find anything.

Reply:
Relevant issue: https://github.com/pypa/pip/issues/3934

Reply:
I am now moving the bin folder to where the RECORD file expects it to be ../../bin and everything is working fine. The checks have become very simple.

Reply:
Yeah I see what you mean, it's odd!